tests/e2e/taqueria-plugin-taquito.spec.ts

```diff
@@ -0,0 +1,346 @@
+            });
+
+            // 4. Verify that contract has been originated to the network
+            await waitForExpect(async () => {
+                const contract = await tezos.contract.at(smartContractHash);
+                expect(contract.address).toBe(smartContractHash);
+            });
+
+        } catch(error) {
+            throw new Error (`error: ${error}`);
+        }
+
+    });
+
+    test('Verify that taqueria taquito plugin can deploy multiple contracts using deploy command', async () => {
+        try {
+            environment = "test";
+            let smartContractOneHash = "";
+            let smartContractTwoHash = "";
+
+            // 1. Copy config.json and two michelson contracts from data folder to artifacts folder under taqueria project
+            execSync(`cp e2e/data/config-taquito-test-environment-multiple-contracts.json ${taqueriaProjectPath}/.taq/config.json`);
+            execSync(`cp e2e/data/hello-tacos.tz ${taqueriaProjectPath}/artifacts/hello-tacos-one.tz`);
+            execSync(`cp e2e/data/hello-tacos.tz ${taqueriaProjectPath}/artifacts/hello-tacos-two.tz`);
+
+            // Sometimes running two deploy commands in a short period of time cause an issue
+            // More details about the error - "counter %x already used for contract %y"
+            // It is applicable only for auto-tests that run very fast
+            // Uses retry mechanism to avoid test to fail
+            await waitForExpect(() => {
+                // 2. Run taq deploy on a network described in "test" environment
+                const stdoutDeploy = execSync(`taq deploy -e ${environment}`, {cwd: `./${taqueriaProjectPath}`}).toString().trim().split(/\r?\n/);
+
+                // 3. Verify that contracts have been originated on the network
+                expect(stdoutDeploy[3]).toContain("hello-tacos-one.tz");
+                expect(stdoutDeploy[3]).toContain("hangzhounet");
+                const [,,contractOneHash] = stdoutDeploy[3].split("│");
+                smartContractOneHash = contractOneHash.trim();
+                expect(smartContractOneHash).toMatch(contractRegex);
+                expect(stdoutDeploy[5]).toContain("hello-tacos-two.tz");
+                expect(stdoutDeploy[5]).toContain("hangzhounet");
+                const [,,contractTwoHash] = stdoutDeploy[5].split("│");
+                smartContractTwoHash = contractTwoHash.trim();
+                expect(contractTwoHash.trim()).toMatch(contractRegex);
+            });
+
+            // 4. Verify that contracts have been originated to the network
+            await waitForExpect(async () => {
+                const contractOne = await tezos.contract.at(smartContractOneHash);
+                expect(contractOne.address).toBe(smartContractOneHash);
+                const contractTwo = await tezos.contract.at(smartContractTwoHash);
+                expect(contractTwo.address).toBe(smartContractTwoHash);
+            });
+
+        } catch(error) {
```

Comment:
Ivan, can we also verify please that the contract addresses are unique?

Reply:
Fixed. Added verification step that hashes are different